tests/cpp/test_memory.cpp

```diff
@@ -1570,6 +1570,395 @@ TEST_F(TMACompileTimeInvalidTest, SwizzleBulkWithNonBulk) {
           "Unsupported expression between the allocation domain and the partitioned IterDomains:")));
 }
 
+// Tests for the examples in doc/dev/tma.md
+class TMADocTest : public TMATest {};
+
+TEST_F(TMADocTest, Figure8a) {
+  GTEST_SKIP() << "TODO: add check for this invalid case.";
```

Comment:
Currently, there is no smem schedule check. So wrong schedule will lead to misaligned address error or silent wrong result at runtime.

Actually, validating the smem schedule is not easy. I don't have a good idea on how to do it yet.